.ci/jenkins/Jenkinsfile.prod.nightly

```
@@ -217,4 +220,15 @@ String calculateArchiveRepoBranch() {
     }
     // e.g., main
     return branch
+}
+
+
+// Parse version from main branch of the given project
+//      * project: in the form of owner/repository
+def parseVersionFromPom(String project) {
+    def pomFilename = "${project.replaceAll("/", "_")}_pom.xml"
+
+    sh "curl https://raw.githubusercontent.com/${project}/main/pom.xml -o ${env.WORKSPACE}/${pomFilename}"
```

```suggestion
    def currentBranch = env.CONFIG_BRANCH ?: env.DEFAULT_CONFIG_BRANCH ?: env.BRANCH_NAME ?: env.GIT_BRANCH
    sh "curl https://raw.githubusercontent.com/${project}/${currentBranch}/pom.xml -o ${pomPath}"
```

I recommend to use current branch instead, input parameters description should be adapted if you follow this approach

---

Good point, I agree with you but we shouldn't use `env.CONFIG_BRANCH` and `env.DEFAULT_CONFIG_BRANCH` because they refer to the build configuration repository. Therefore I used:
```
def currentBranch = getCurrentBranch() ?: env.GIT_BRANCH
```